File: twoSum/twoSum.go
package problems

// Time:  O(n)
// Space: O(n)

func twoSum(nums []int, target int) []int {
	lookUP := make(map[int]int, len(nums))
	for i, num := range nums {
		if j, ok := lookUP[target-num]; ok {
			return []int{j, i}
		}
		lookUP[nums[i]] = i
	}
	return nil
}

File: twoSum/twoSum.py
#!/usr/bin/env python

class problems(object):
    def twoSum(self,nums,target):
        lookUP = {}
        for i,num in enumerate(nums):
            if target - num in lookUP:
                return [lookUP[target - num], i]
            lookUP[num] = i
